qst/src/main/java/io/deephaven/qst/type/BoxedTypeVisitorAdapter.java

```diff
@@ -0,0 +1,53 @@
+package io.deephaven.qst.type;
+
+import io.deephaven.qst.type.PrimitiveType.Visitor;
+
+import java.util.Objects;
+
+class BoxedTypeVisitorAdapter<R> implements BoxedType.Visitor<R> {
+    private final PrimitiveType.Visitor<R> delegate;
+
+    public BoxedTypeVisitorAdapter(Visitor<R> delegate) {
+        this.delegate = Objects.requireNonNull(delegate);
+    }
+
+    @Override
+    public R visit(BoxedBooleanType booleanType) {
+        return delegate.visit(booleanType.primitiveType());
+    }
```

Comment:
Do we need this much code? Why not just walk the delegate from the BoxedType's primitive?

Reply:
Yes - that's a good pattern, and cuts down on the code. I thought I would need these adapters, but ultimately didn't (and used the pattern like you suggested).